tests/x11/krunner.pm

```diff
@@ -0,0 +1,35 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test the advanced features of krunner, e.g. autocompletion
+# Maintainer: Oliver Kurz <okurz@suse.de>
+
+use base 'x11test';
+use strict;
+use warnings;
+use testapi;
+use x11utils 'desktop_runner_hotkey';
+
+sub run {
+    wait_screen_change { send_key desktop_runner_hotkey };
+    type_string 'echo "Hello World"';
+    assert_screen 'krunner-hello_world';
+    send_key 'ret';
+    # on wayland krunner sometimes vanishes or crashes, e.g. after the second
```

Comment:
I've never seen that, at least not outside of openQA.

Reply:
And that is basically the whole point of this PR and the according ticket https://progress.opensuse.org/issues/35589 :)

https://openqa.opensuse.org/tests/871386/file/video.ogv#t=32.08,32.14 shows a good example. Play it and you see that the first characters are correctly typed, then suddenly the desktop runner vanishes and reappears but without the previously typed characters.

Reply:
It's neither a crash nor a relayout (which happens as the window size changes). I'll try to reproduce that somehow (but already tried multiple times, without "success").